components/calendar/src/cal/hijri.rs

```diff
@@ -837,39 +730,341 @@ impl Calendar for HijriUmmAlQura {
     }
 }
 
-impl DatafulHijri for HijriUmmAlQuraMarker {
-    fn info_for_rd(&self, rd: RataDie) -> HijriYearInfo<Self> {
-        let (y, _m, _d) =
-            calendrical_calculations::islamic::tabular_islamic_from_fixed(rd, ISLAMIC_EPOCH_FRIDAY);
-        self.info_for_year(y)
-    }
-
-    fn info_for_year(&self, extended_year: i32) -> HijriYearInfo<Self> {
-        HijriYearInfo::from_parts(
-            extended_year,
-            core::array::from_fn(|i| {
-                HijriTabular::days_in_provided_month(extended_year, i as u8 + 1) == 30
-            }),
-            calendrical_calculations::islamic::fixed_from_tabular_islamic(
-                extended_year,
-                1,
-                1,
-                ISLAMIC_EPOCH_FRIDAY,
-            ),
-            *self,
-        )
+impl PrecomputedDataSource<HijriYearInfo> for HijriUmmAlQura {
+    fn load_or_compute_info(&self, year: i32) -> HijriYearInfo {
+        #[rustfmt::skip]
+        const DATA: [PackedHijriYearInfo; 1601 - 1300] = {
+            use calendrical_calculations::iso::fixed_from_iso;
+            let l = true; // long
+            let s = false; // short
+            [
+                PackedHijriYearInfo::new(1300, [l, s, l, s, l, s, l, s, l, s, l, s], fixed_from_iso(1882, 11, 12)),
+                PackedHijriYearInfo::new(1301, [l, l, s, l, s, l, s, l, s, l, s, s], fixed_from_iso(1883, 11, 1)),
+                PackedHijriYearInfo::new(1302, [l, l, l, s, l, l, s, s, l, s, s, l], fixed_from_iso(1884, 10, 20)),
+                PackedHijriYearInfo::new(1303, [s, l, l, s, l, l, s, l, s, l, s, s], fixed_from_iso(1885, 10, 10)),
+                PackedHijriYearInfo::new(1304, [s, l, l, s, l, l, l, s, l, s, l, s], fixed_from_iso(1886, 9, 29)),
+                PackedHijriYearInfo::new(1305, [s, s, l, l, s, l, l, s, l, l, s, s], fixed_from_iso(1887, 9, 19)),
```

Comment:
Issue: This is cute, but my enthusiasm around `const` has fizzled a bit with all the real problems we've been having with baked data compile times. I think I'd prefer putting the raw data here and having a test that asserts it is consistent with the readable version. Then we don't need to worry about const.

Reply:
I find this comment pretty insulting, I have designed this to be readable and maintainable (for the next time we need to figure out why a UAQ date doesn't match), not for it to be "cute". This has no measurable impact on compile time.

Reply:
The word "cute" was intended to be positive. I like what you did here except that I'm not convinced it is worth having all the extra const stuff to make it work.

Reply:
I'm still not 100% convinced, but I really do like how this code reads, so if you say that it doesn't increase compile times, I'm okay landing this so long as my other open comments are resolved.